### PR TITLE
chore: docusaurus@2.0.0-beta.15

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

-   [x] Addresses an existing issue: starts on #4361
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

Brings in the latest Docusaurus version for https://github.com/facebook/docusaurus/pull/6288. Prerequisite to #4362.

### CODE REVIEW

File: packages/website/src/components/Playground.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React, { useReducer, useState } from 'react';
 import type Monaco from 'monaco-editor';
 import clsx from 'clsx';
-import useThemeContext from '@theme/hooks/useThemeContext';
+import { useColorMode } from '@docusaurus/theme-common';
```
Comment (JoshuaKGoldberg) [inline packages/website/src/components/Playground.tsx:4]: @Josh-Cena am I doing this right?

I recall you mentioning that we could probably un-swizzle some components? If there's a big refactor I'd be down to take that as a followup.

Comment (Josh-Cena) [reply packages/website/src/components/Playground.tsx:4]: Hmmm... Did I? :D

And yes, you are doing it right.

Comment (JoshuaKGoldberg) [reply packages/website/src/components/Playground.tsx:4]: haha awesome, thanks!

**Review by bradzacher — APPROVED** [review]
image

**Pull request merged**